Breaking-change announcement for MQTT CN validation enforcement (#3941)
SoftwareAG/c8y-docs: content/change-logs/platform-services/mqtt-service-0.9.x-cn-validation-announcement.md

---
date: 2025-11-XX

> title: MQTT Service will enforce CN validation for certificate-authenticated clients

Reviewer: ```suggestion
title: MQTT Service will enforce Common Name validation for certificate-authenticated clients
```

change_type:
  - value: change-inv-3bw8e
    label: Announcement
product_area: Platform services
component:
  - value: component-LcWEQW5gs
    label: MQTT
build_artifact:
  - value: tc-hc5Tfixeqqei
    label: mqtt-service
issue: MTM-65167
---

### Introduction

To strengthen identity assurance for certificate-authenticated MQTT clients, the {{< product-c8y-iot >}} [MQTT Service](/device-integration/mqtt-service/) will begin enforcing **Common Name (CN) validation** during client certificate authentication.

Currently, the MQTT Service accepts certificates where the CN does not match the MQTT client ID.  

> This will change: the CN must correspond to the client ID used during connection, improving device-to-certificate integrity and reducing the risk of certificate misuse.

Reviewer: ```suggestion
After this change, the CN must match the client ID used during connection.
This tight binding of certificates to devices will significantly reduce the risk of certificate misuse.
```

### What is changing?

When an MQTT client connects using certificate-based authentication:

* The **CN in the certificate must match the MQTT client ID**.
* SmartREST-style identifiers are supported, including:
  * `CN == <clientId>`

>   * `CN == "d:<clientId>"`

Reviewer: Suggestion: Rework this to be really clear that there are only 2 allowed CNs for any given client id, and that the `d:<clientID>` form is only to support legacy SmartREST devices migrating to the MQTT Service. That format must not be used for new devices.

Only clients using certificate-based authentication are affected.  
No changes apply to clients using other authentication mechanisms.

### Impact on existing MQTT clients

This is a **breaking change**.  
Devices using certificates whose CN does not align with the MQTT client ID will fail authentication once enforcement begins.

Customers should verify and update their certificate issuance processes during the grace period.

> For support, please contact product support.

Reviewer: Thought: I think we normally say something like "If you have any questions or concerns, please contact Cumulocity support" and link to the "Contacting Support" section in the docs.

### Roll-out plan

> To allow a smooth transition, CN validation will be introduced **four weeks after this announcement**.

Reviewer: ```suggestion
To allow a smooth transition, CN validation will be introduced no sooner than **four weeks after this announcement**.
```

issue: MTM-65167
---

### Introduction

To strengthen identity assurance for certificate-authenticated MQTT clients, the {{< product-c8y-iot >}} [MQTT Service](/device-integration/mqtt-service/) will begin enforcing **Common Name (CN) validation** during client certificate authentication.

Currently, the MQTT Service accepts certificates where the CN does not match the MQTT client ID.  
After this change, the CN must match the client ID used during connection.
This tight binding of certificates to devices will significantly reduce the risk of certificate misuse.

### What is changing?

When an MQTT client connects using certificate-based authentication, **the Common Name (CN) in the certificate must match the MQTT client ID**.

Only **two CN formats** will be accepted for any given client ID:
1. `CN == <clientId>` - the standard and **required** format for all new devices.
2. `CN == d:<clientId>` - supported **only for legacy SmartREST devices** migrating to the MQTT Service. This format must **not** be used for new devices.

If the CN does not match one of these two allowed forms for the client ID presented during connection, the client will fail authentication.

> Only certificate-authenticated clients are affected; all other authentication methods remain unchanged.

Reviewer: Issue: Not sure how we all missed this earlier, but isn't this the wrong way around? The _client id_ can be `<deviceid>` or `d:<deviceid>` but the _CN_ will be `<deviceid>` in both cases, right?

Author: Done.

### Impact on existing MQTT clients

This is a **breaking change**.  
Devices using certificates whose CN does not align with the MQTT client ID will fail authentication once enforcement begins.

Customers should verify and update their certificate issuance processes during the grace period.

Please contact [Cumulocity Support](/additional-resources/contacting-support/) if you have any questions or concerns about these changes.

### Roll-out plan

> To allow a smooth transition, CN validation will be introduced no sooner than **four weeks after this announcement**.

Reviewer 2: Should we say four weeks? According to our policy (https://cumulocity.com/docs/service-terms/compatibility-policy/) that should be 6 months. Maybe we should say that because MQTT Service is in public preview, changes will be introduced once it goes GA or something similar (just to avoid any questions from customers).

Reviewer: > Should we say four weeks? According to our policy (https://cumulocity.com/docs/service-terms/compatibility-policy/) that should be 6 months. Maybe we should say that because MQTT Service is in public preview, changes will be introduced once it goes GA or something similar (just to avoid any questions from customers).

Good thought. Maybe handle this with the warnings like we used for previous release notes: https://cumulocity.com/docs/change-logs/?component=.component-mqtt#mqtt-service-0.9.21-pulsar-client-preview? Customers have freaked out when they thought we were making breaking changes to Core MQTT so we should probably have these anyway (and I should have remembered that)

Author: Done.

---
date: 2025-11-XX
title: MQTT Service will enforce Common Name validation for certificate-authenticated clients
change_type:
  - value: change-inv-3bw8e
    label: Announcement
product_area: Platform services
component:
  - value: component-LcWEQW5gs
    label: MQTT
build_artifact:
  - value: tc-hc5Tfixeqqei
    label: mqtt-service

> issue: MTM-65167

Reviewer 3: Version is missing.

Author: Do announcements need a version label? I recall that in the past it was not the case. 

As per this document [Release Notes creation process 2.0](https://docs.google.com/document/d/1pU2VIdJUIX1mKKRXF1GNHFZzKtCmTcRz), it states, "The date to be used for an announcement can be the current date. The version field can be left empty."

Reviewer 3: Ah yes, in this case you are right, it is not yet happening. Announcements are used quite differently, so sometimes they "announce" something that is immediately implemented.
